arroyo-rpc/src/types.rs

```diff
@@ -13,6 +13,27 @@ pub struct ValidatePipelinePost {
     pub udfs: Option<Vec<Udf>>,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct PipelineGraphValidationResult {
+    pub graph: Option<PipelineGraph>,
+    pub errors: Option<Vec<String>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct UdfsValidationResult {
+    pub udfs: Option<Vec<Udf>>,
```

Comment:
The logic here is that `udfs` (and `PipelineGraphValidationResult::graph`) are only Some if the validation was successful and there are no errors, i.e. only one of these fields is Some. I originally had this as an enum like this:

```
pub enum UdfsValidationResult {
    Ok(Vec<Udf>),
    Error(Vec<String>),
}
```

but this creates a bit of a weird API json schema, since there isn't really a standard way of representing in json the Rust style of enums variants with values.

Reply:
That still doesn't really make sense to me as an api consumer. Why are we returning the udfs at all?

Reply:
I guess the weird thing is that we're using rust-style errors for an http endpoint, where you'd expect an error response in the error case.

Reply:
Anyways I think this is fine for now but we should probably think about how to more naturally express this in REST for the next version

Reply:
> Why are we returning the udfs at all

My thinking is that POST endpoints are typically supposed to return a representation of the object that was created, and even though we're not creating an object with this request, we can still return the thing that was validated, kind of just like saying _this was the thing we validated_.

> does it make sense to distinguish between the empty vec case and none cases for errors

This is debatable but I think most people would [say](https://stackoverflow.com/questions/1969993/is-it-better-to-return-null-or-empty-collection) an empty vector is preferred because it avoids the class of null/undefined related errors. 

> I guess the weird thing is that we're using rust-style errors for an http endpoint, where you'd expect an error response in the error case.

I'm not sure that's the proper use of HTTP error codes, since the _errors_ we're talking about here originate from the code provided by the client, not on the server. I think 200 responses are still appropriate when the thing we're validating ends up being invalid.